### PR TITLE
unpack_strategy: add zstd strategy

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This PR adds support for `.zst` and `.tar.zst` archives formats. 

So far, there's only one formula in homebrew-core that uses it: [`goredo`](https://github.com/Homebrew/homebrew-core/blob/bb2d76691931d322f7cf25f1ced0d222074c95f0/Formula/goredo.rb). After this PR we could simplify its installation.

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Nice work!

File: Library/Homebrew/dependency_collector.rb [hunk 1]
```diff
@@ -170,13 +178,14 @@ def resource_dep(spec, tags)
...
     case File.extname(url)
-    when ".xz"          then xz_dep_if_needed(tags)
-    when ".zip"         then unzip_dep_if_needed(tags)
-    when ".bz2"         then bzip2_dep_if_needed(tags)
-    when ".lha", ".lzh" then Dependency.new("lha", tags)
-    when ".lz"          then Dependency.new("lzip", tags)
-    when ".rar"         then Dependency.new("unrar", tags)
-    when ".7z"          then Dependency.new("p7zip", tags)
+    when ".xz"              then xz_dep_if_needed(tags)
+    when ".zst", ".tar.zst" then zstd_dep_if_needed(tags)
```
Comment (Bo98) [inline Library/Homebrew/dependency_collector.rb:182]: Does `.tar.zst` ever get matched here?

Comment (bayandin) [reply Library/Homebrew/dependency_collector.rb:182]: You're right, I've confused `File::extname` with `Pathname#extname` here.
Thanks!

File: Library/Homebrew/unpack_strategy/tar.rb [hunk 2]
```diff
@@ -42,9 +43,9 @@ def extract_to_dir(unpack_dir, basename:, verbose:)
         tar_path = path
 
         if DependencyCollector.tar_needs_xz_dependency? && Xz.can_extract?(path)
```
[the remark this answers is not in the document]
Comment (bayandin) [reply Library/Homebrew/unpack_strategy/tar.rb]: It should! Good catch!

**Review by Bo98 — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Great work as usual @bayandin!

**Pull request merged**